uips/uip-7.md

```diff
@@ -0,0 +1,133 @@
+| discussions-to | https://forum.penumbra.zone/t/pre-uip-transparent-addresses/140 |
+| status | Draft |
+| type | Standards |
+| consensus | Yes |
+| created | 2024-12-10 |
+
+## Abstract
+
+This UIP introduces transparent addresses ("t-addresses"), a 32-byte Bech32 address format designed for maximum compatibility with external systems while maintaining convertability to full Penumbra addresses.
+
+## Motivation
+
+Recent compatibility issues with external systems (e.g., Noble's USDC integration) have highlighted the need for a maximally compatible address format. While the correct long-term solution is full support for Penumbra's native addresses, a shorter address format can serve as an immediate solution and future-proof escape hatch for similar issues. Penumbra's native addresses have two major differences from most other addresses in the Cosmos ecosystem, in that:
+
+1. They are longer: 144 bytes.
+2. They use Bech32m encoding, instead of Bech32.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Address Format
+
+The current Penumbra address format consists of a:
+* 16-byte diversifier $d$,
+* 32-byte transmission key $pk_d$, and
+* 32-byte clue key $ck_d$.
+
+The diversifier is generated by encrypting a 16-byte _address index_ which
+consists of:
+* 4 byte account index, the default account index being 0,
+* 12 byte randomizer
+
+The diversifier $d$ is then used to derive the diversified basepoint $B_d$ using
+decaf377 hash-to-group $H_{\mathbb{G}}^{d}$:
+
+$B_d = H_{\mathbb{G}}^{d}​(d)$
+
+This diversified basepoint is used to to derive the transmission key $pk_d$:
+
+$pk_d = [ivk] B_d$
+
+The new address format, the *transparent address*, consists only of the 32 byte transmission key. It is encoded using the `tpenumbra` human-readable prefix.
+
+The transmission key for the transparent address is derived using the diversifier
+corresponding to all zero bytes. Note that during scanning, the wallet scans
+for incoming messages using the Incoming Viewing Key (IVK), and thus detects
+all messages for all transmission keys corresponding to that IVK.
+
+### Transparent Address Conversion Rules
+
+When converting a transparent address to a full Penumbra address:
+1. The diversifier MUST be set to 16 zero bytes
+2. The transmission key MUST be set to the transmission key of the transparent address
+3. The clue key MUST be set to the identity element
+
+### Diversifier Decryption
+
+The diversifier decryption is modified as follows:
+
+The null ciphertext [0u8; 16] is defined to correspond to the address index
+corresponding to the default account index 0, with no randomizer.
+
+The implications of this are that there are two valid diversifiers for the
+default account:
```

Comment:
More than two, right, since you have

- `2^96` diversifiers that (normally) decrypt to address indexes with the first four bytes all zeros
- `1` diversifer (all zeros) that (special-case) decrypts to the zero address index

It would probably be helpful in this section to note that although `[0u8; 16]` is a valid ciphertext for some address index `a`, a user is astronomically unlikely to ever select this address index in normal use (as is noted later in the doc).